src/beatstar/database.rs

```diff
@@ -225,6 +284,54 @@ pub fn beatstar_retrieve_database() -> anyhow::Result<&'static BeatStarDataFile>
     Ok(bsf_mutex)
 }
 
+fn beatstar_update_database_from_file(file_path: &str) -> anyhow::Result<()> {
+    initialize_log();
+    // Read zip from path
+    let bytes: Vec<u8> = std::fs::read(file_path)?;
+    
+    let mut stopwatch = Stopwatch::start_new();
+
+    let body: Vec<BeatStarSong> = beatstar_zip_content_from_bytes(bytes)
+        .context("Failed to parse scrapped beat saver data zip.")?;
+        event!(
+            Level::INFO,
+            "Parsed beat file into json data in {0}ms",
+            stopwatch.elapsed().as_millis()
+        );
+
+        // Get data inside file and map it
+        let parsed_data = parse_beatstar(body);
+
+        let json_size = parsed_data.songs.iter()
+            .map(|(str, diff)| unsafe { CStr::from_ptr(str.string_data).to_bytes().len() } + std::mem::size_of_val(&diff))
+            .reduce(|acc, i| acc + i).unwrap_or(0);
+
+        BEAT_STAR_FILE.get_or_init(|| parsed_data);
```

Comment:
Instead of passing the json as a parecerse, move all the logic into the lambda. In the current state, this will read a file and parse the JSON unnecessarily if the memory cache is already initialized.